src/main/java/me/crespel/runtastic/converter/ExportConverter.java

```diff
@@ -9,60 +9,125 @@
-		path = normalizeSportSessionPath(path);
+	public List<SportSession> listSportSessions(File path, boolean full) throws FileNotFoundException, IOException {
+		path = normalizeExportPath(path, SPORT_SESSIONS_DIR);
 		List<SportSession> sessions = new ArrayList<>();
 		File[] files = path.listFiles(file -> file.getName().endsWith(".json"));
 		for (File file : files) {
-			sessions.add(parser.parseSportSession(file));
+			sessions.add(parser.parseSportSession(file,full));
 		}
 
 		Collections.sort(sessions);
 		return sessions;
 	}
 
+	public User getUser(File path) throws FileNotFoundException, IOException {
+		return parser.parseUser(new File(normalizeExportPath(path, USER_DIR), "user.json"));
+	}
+
 	public SportSession getSportSession(File path, String id) throws FileNotFoundException, IOException {
-		path = normalizeSportSessionPath(path);
-		return parser.parseSportSession(new File(path, id + ".json"));
+		return parser.parseSportSession(new File(normalizeExportPath(path, SPORT_SESSIONS_DIR), id + ".json"), true);
+	}
+
+	public SportSession getSportSessionWithPhoto(File path, String photoid) throws FileNotFoundException, IOException {
+		String sessionid = null;
+
+		File photofile = new File(normalizeExportPath(path, PHOTOS_DIR), photoid + ".jpg");
+		if( photofile.exists() ) {
+			// photo file found ...
+
+			ImagesMetaData image = parser.parseImagesMetaData(new File(normalizeExportPath(path, PHOTOS_META_DATA_DIR), photoid + ".json"));
+			if( image != null ) {
+				// photo meta data file found ...
+
+				// search trough sport session album data, to find sport session related to the photo
+				File[] files = normalizeExportPath(path, PHOTOS_SPORT_SESSION_ALBUMS_DIR).listFiles(file -> file.getName().endsWith(".json"));
+				for( File file : files ) {
+					try {
+						SportSessionAlbums mysessionalbum = parser.parseSportSessionAlbumsData(file);
+						if (mysessionalbum.getPhotosIds().contains(photoid)) {
+							// Sport session id found within sport session albums
+							sessionid = mysessionalbum.getId();
+						}
+					} catch (IOException e) {
+						throw new RuntimeException(e);
+					}
+				}
+			}
+		}
+
+		return getSportSession(path, sessionid);
+	}
+
+	public List<SportSession> convertSportSessions(File path, String format) throws FileNotFoundException, IOException {
+		File[] files = normalizeExportPath(path, SPORT_SESSIONS_DIR).listFiles(file -> file.getName().endsWith(".json"));
+		List<SportSession> sessionlist = new ArrayList<>();
+		Arrays.asList(files).parallelStream().forEach(file -> {
+			try {
+				SportSession session = parser.parseSportSession(file, true);
+				if (session.getGpsData() != null || session.getHeartRateData() != null || session.getGpx() != null) {
+					GpxType gpx = (GpxType) mapper.mapSportSession(session, format);
+					session.setGpx(gpx);
+				}
+				sessionlist.add(session);
+				System.out.print(".");

```

Comment:
I understand it's interesting to show progress to the user, but this class shouldn't deal with user interaction (only the main class `RuntasticExportConverter` should do it). A better way to handle this separation of concern would be to pass an optional functional interface to `ExportConverter` that it can use to report progress back to its caller.

This is not mandatory to get this PR approved, just pointing out a better alternative ;-)

Reply:
Agreed! I removed screen output. It was anyway more for me during develpment (debug output).